server/routes/prisonerChangeIncentiveLevelDetails.test.ts

```diff
@@ -0,0 +1,298 @@
+  offenderNo: prisonerNumber,
+  agencyId: 'MDI',
+  bookingId,
+  firstName: 'John',
+  lastName: 'Smith',
+  assignedLivingUnit: {
+    agencyId: 'MDI',
+    locationId: 1,
+    description: '123',
+    agencyName: '123',
+  },
+}
+
+const offenderDetails: OffenderSearchResult = {
+  bookingId,
+  prisonerNumber,
+  firstName: 'John',
+  lastName: 'Smith',
+  prisonId: 'MDI',
+  prisonName: 'Moorland',
+  cellLocation: '123',
+}
+
+const userCaseload: UserCaseload = {
+  activeCaseload: {
+    id: 'MDI',
+    name: 'MDI',
+  },
+  caseloads: [
+    {
+      id: 'MDI',
+      name: 'MDI',
+    },
+  ],
+}
+
+const prisonApi = PrisonApi.prototype as jest.Mocked<PrisonApi>
+const offenderSearch = OffenderSearchClient.prototype as jest.Mocked<OffenderSearchClient>
+const incentivesApi = IncentivesApi.prototype as jest.Mocked<IncentivesApi>
+const nomisUserRolesApi = NomisUserRolesApi.prototype as jest.Mocked<NomisUserRolesApi>
+beforeEach(() => {
+  prisonApi.getPrisonerDetails.mockResolvedValue(prisonerDetails)
+  prisonApi.getFullDetails.mockResolvedValue(prisonerDetails)
+  offenderSearch.getPrisoner.mockResolvedValue(offenderDetails)
+  incentivesApi.getIncentiveSummaryForPrisoner.mockResolvedValue(incentiveSummaryForBooking)
+  incentivesApi.getPrisonIncentiveLevels.mockResolvedValue(samplePrisonIncentiveLevels)
+  nomisUserRolesApi.getUserCaseloads.mockResolvedValue(userCaseload)
+
+  app = appWithAllRoutes({})
+})
+
+afterEach(() => {
+  jest.resetAllMocks()
+})
+describe('GET /incentive-reviews/prisoner/change-incentive-level', () => {
+  it('should NOT allow user to update incentive level without role', async () => {
+    return request(app)
+      .get(`/incentive-reviews/prisoner/${prisonerNumber}/change-incentive-level`)
+      .set('authorization', `bearer ${tokenWithMissingRole}`)
+      .expect(res => {
+        expect(res.redirect).toBeTruthy()
+      })
+  })
+
+  it('should render the correct template with the correct data', async () => {
+    return request(app)
+      .get(`/incentive-reviews/prisoner/${prisonerNumber}/change-incentive-level`)
+      .set('authorization', `bearer ${tokenWithNecessaryRole}`)
+      .expect('Content-Type', /html/)
+      .expect(res => {
+        expect(res.text).toContain('Smith, John')
+        expect(res.text).toContain('Record an incentive level')
+        expect(res.text).toContain('Standard (current level)')
+        expect(res.text).toContain('Reason for recording')
+        expect(res.text).toContain('Save')
+      })
+  })
+
+  it('should return 404 if prisoner is not found', () => {
+    const error: SanitisedError = {
+      name: 'Error',
+      status: 404,
+      message: 'Not Found',
+      stack: 'Not Found',
+    }
+    offenderSearch.getPrisoner.mockRejectedValue(error)
+    return request(app)
+      .get(`/incentive-reviews/prisoner/${prisonerNumber}/change-incentive-level`)
+      .set('authorization', `bearer ${tokenWithNecessaryRole}`)
+      .expect(200)
+  })
+})
```

Comment:
i think this test isn't quite right. from what i can tell, the "record incentive level" page doesn't call offender search so really
1) offender search mocking ought to be removed from this file
2) the 404 could instead be done on `prisonApi.getPrisonerDetails` and then i think you can once again have something like this, as you did in the past
```
      .expect(404)
      .expect(res => {
        expect(res.text).toContain('Page not found')
        expect(res.text).not.toContain('John, Smith')
      })
```